### ISSUE TITLE
Flyspell not checking yml data

### ISSUE THREAD

**Issue by tranner** [jrblevin/markdown-mode#560]
## Expected Behavior

The yml metadata at the top of the file contains important information including titles as well as code. Some of these should be spell checked by default. I would at least think it would be good to check for strings in quotes (i.e. "title") and spell check those.

## Actual Behavior

The whole yml block is treated as code so nothing here picked up by flycheck leading to spelling mistakes in my title :(

## Steps to Reproduce

1. Download two files. `test.el.txt`  and `test.md.txt`  and remove .txt extension (needed for uploading?!):
[test.el.txt](https://github.com/jrblevin/markdown-mode/files/5532143/test.el.txt)
[test.md.txt](https://github.com/jrblevin/markdown-mode/files/5532144/test.md.txt)

2. Run `emacs -Q -l test.el test.md`
3. See asdasdasd in the main body highlighted as spelling error but not in the title.

Example output:

![output](https://user-images.githubusercontent.com/1355684/98973141-448ea880-250b-11eb-864c-57b9340bcb3f.png)

## Software Versions

- Markdown Mode: markdown-mode, version 2.4-dev
- Emacs: GNU Emacs 27.1.50 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2020-08-21
- OS: CentOS Linux release 7.8.2003 (Core)

**Comment by syohex** [comment]
```diff
diff --git a/markdown-mode.el b/markdown-mode.el
index a1e37e0..a92a5c8 100644
--- a/markdown-mode.el
+++ b/markdown-mode.el
@@ -2150,7 +2150,8 @@ Depending on your font, some reasonable choices are:
 Used for `flyspell-generic-check-word-predicate'."
   (save-excursion
     (goto-char (1- (point)))
-    (if (or (markdown-code-block-at-point-p)
+    (if (or (and (markdown-code-block-at-point-p)
+                 (not (markdown-text-property-at-point 'markdown-yaml-metadata-section)))
             (markdown-inline-code-at-point-p)
             (markdown-in-comment-p)
             (markdown--face-p (point) '(markdown-reference-face
```

![a](https://user-images.githubusercontent.com/554281/99029473-dfdc5800-25b5-11eb-96c0-bae054514807.png)

I suppose above patch fixes your issue. However I don't understand yet why original code ignores spell checking in YAML meta header.

**Comment by tranner** [comment]
Thanks - this seems to solve the problem at hand at least.

I end up with lots of spelling errors for the metadata keys too but I'd rather have too many errors highlighted rather than too few for now.

**Comment by syohex** [comment]
I will apply above patch after merging #559 for avoiding code conflict.

**Comment by syohex** [comment]
I have merged #561. Please check latest version.

**Comment by tranner** [comment]
Thanks again - all fixed :+1:
